MonMaraicher/Features/Map/MapViewModel.swift

```diff
@@ -34,7 +35,9 @@ final class MapViewModel: ObservableObject {
     init(farmerService: FarmerService) {
         self.farmerService = farmerService
         self.imageSystemNameSearchButton = "magnifyingglass"
-        loadFarmers()
+        Task {
+            await loadFarmers()
```

Comment:
In SwiftUI you can run a task using the modifier `.task {...}`. So you could either make you loadFarmers internal instead of private and call it from your view. It a easy one but not the best as you can imagine.

I would advice you to load data according to the view state and not in your viewModel init. Imagine you init you viewModel and not your view right away, you will load data too early for instance.

I would load data in the onAppear of your viewModel to be sure that the view has been loaded